Updated docs to remove background icons examples;

seangates opened the pull request:
## Description
- made sure the foreground (SVG) icon was shown as "preferred" in the docs
- removed the demo app example for background icons

## Context
We want to promote foreground (SVG) icons as the way to correctly use icons with the core components. Very luckily, none of our components use the background icons. Hurray!

## References
Fixes #500

seangates commented:
@ianmcburnie That reference is to the `<ebay-icon>` tag, not the Skin bundle for `@ebay/skin/icon`.

ianmcburnie commented:
Oh yes. Oops. Seems like my brain was switched off. Thanks Sean.

ianmcburnie commented:
@seangates Maybe this is the line I meant:

https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-icon/browser.json#L5

ianmcburnie commented:
As discussed offline, let's target this for a 1.4.0 branch and release.

ianmcburnie commented:
By the way, the intent of #500 isn't just to remove background icons from our docs, but all internal references too.

seangates commented:
@ianmcburnie Yup. I just haven't gotten to your comment. Working on it now. I will create a new 1.4.0 branch and re-point my changes to that target branch.

seangates commented:
`"path": "@ebay/skin/icon/foreground"` does not appear to be allowed right now.

ianmcburnie commented:
Did you update to skin 6.3?

seangates commented:
@ianmcburnie This should be good now, and pointed to `1.4.0`.

ianmcburnie commented on src/components/ebay-button/index.js:
```diff
@@ -38,22 +43,34 @@ function getInitialState(input) {
 
-    if (!isBadged && (size === 'small' || size === 'large')) {
-        classes.push(`${mainClass}--${size}`);
+    if (size === 'small' || size === 'medium' || size === 'large') {
```
Are these code changes intentionally part of this icon related PR?

seangates commented:
@ianmcburnie Fixed!

seangates commented on package.json:
```diff
@@ -58,7 +58,7 @@
   "devDependencies": {
     "@ebay/browserslist-config": "^1.0.0",
-    "@ebay/skin": "6.2.0",
+    "@ebay/skin": "6.3.0",
```
This is a `devDependency` though, and wouldn't be used when pulled into Inception, correct?

ianmcburnie replied:
peerDependency is also pointing to 6.2.0

seangates commented:
@ianmcburnie I have updated the package versions for Skin.

Also, I went ahead and updated the icon example with a lot less icons (there was literally no need), so please note that change.

ianmcburnie commented:
I think we should keep at least one example of background icon.

seangates commented:
There’s an example in the README, which I did not remove, and will show up in the marketplace.

seangates commented:
@ianmcburnie I was able to find the additional dependency paths, and everything should be right. I no longer see the CSS file being linked in the demo app.
